feat: runtime metrics (#375)
signalfx/splunk-otel-js: docs/advanced-config.md

- `instrumentations`: Can be used to enable additional instrumentation packages.

- `captureHttpRequestUriParams`: Either a list of keys (case-sensitive) of HTTP query parameters to capture or a function that gets invoked with the current span and query parameters to set a custom span attribute. When using the former, parameters are set as span attributes as `http.request.param.${key}`. Attribute keys are normalized at capture time, meaning `.` is replaced with `_` to avoid any attribute namespacing issues.

### Metrics

| Environment variable<br>``startMetrics()`` argument             | Default value           | Support | Notes
| --------------------------------------------------------------- | ----------------------- | ------- | ---
| `SPLUNK_METRICS_ENABLED`<br>`enabled`                           | `false`                 | Experimental | Enabled metrics export. See [metrics documentation](metrics.md) for more information.
| `SPLUNK_METRICS_ENDPOINT`<br>`endpoint`                         | `http://localhost:9943` | Experimental | The SignalFx metrics endpoint to send to.

> | `SPLUNK_METRICS_EXPORT_INTERVAL`<br>`exportInterval`            | `5000`                  | Experimental | The interval, in milliseconds, of metrics collection and exporting.

Reviewer: Perhaps an extra column for the `startMetrics` argument would make it easier to scan the table.

Reviewer: s/SignalFx/Splunk

Author: Added a link to SignalFx client. Followed the existing doc style and `startTracing` did not have the extra column either, should I split both of them to separate column?

#### Additional `startMetrics` config options

> - `signalfx`: A JS object with optional `client` and `dimensions` fields. If you have already setup a SignalFx client with custom configuration, it is possible use this for sending instead of creating, configuring a new one. `dimensions` object adds a pre-defined dimension for each datapoint. The format for `dimensions` is `{key: value, ...}`.

Reviewer: ```suggestion
- `signalfx`: A JS object with optional `client` and `dimensions` fields. If you have already setup a Splunk client with custom configuration, you can use this for sending instead of creating, configuring a new one. `dimensions` object adds a pre-defined dimension for each datapoint. The format for `dimensions` is `{key: value, ...}`.
```

Can we rename the setting to "splunk"?

Author: Hmm, explicitly wanted to make it clear that we are dealing with [SignalFx Node.js client](https://github.com/signalfx/signalfx-nodejs). What do you think about removing the `signalfx` field completely? And replace it with 2 fields: `client` and `dimensions`?

Reviewer: The old library? Then it makes sense. Just wondering about backward compatibility if the switch to OTel metrics in the future (see Owais' comments).

Author: I added a link to the `SignalFx client` there and mentioned that metrics will change once OpenTelemetry becomes available.
